src/jobs/check-database.ts

```diff
@@ -0,0 +1,64 @@
+/**********************************************************************************
+ * MIT License                                                                    *
+ *                                                                                *
+ * Copyright (c) 2021 Hyperjump Technology                                        *
+ *                                                                                *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy   *
+ * of this software and associated documentation files (the "Software"), to deal  *
+ * in the Software without restriction, including without limitation the rights   *
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell      *
+ * copies of the Software, and to permit persons to whom the Software is          *
+ * furnished to do so, subject to the following conditions:                       *
+ *                                                                                *
+ * The above copyright notice and this permission notice shall be included in all *
+ * copies or substantial portions of the Software.                                *
+ *                                                                                *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR     *
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,       *
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE    *
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER         *
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,  *
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE  *
+ * SOFTWARE.                                                                      *
+ **********************************************************************************/
+
+import path from 'path'
+import fs from 'fs'
+import { getConfig } from '../components/config'
+import {
+  deleteFromAlerts,
+  deleteFromNotifications,
+  deleteFromProbeRequests,
+} from '../components/logger/history'
+import { Config } from '../interfaces/config'
+const dbPath = path.resolve(process.cwd(), 'monika-logs.db')
+
+export function check_db_size() {
+  const config = getConfig()
+  deleteData(config)
+}
+
+async function deleteData(config: Config) {
+  const { db_limit } = config
+  const stats = fs.statSync(dbPath)
+
+  if (!db_limit.max_db_size) {
+    return
+  }
+
+  if (!db_limit.deleted_data) {
+    return
+  }
+
+  if (stats.size > db_limit.max_db_size) {
+    const probe_res = await deleteFromProbeRequests(db_limit.deleted_data)
+    const notif_res = await deleteFromNotifications(db_limit.deleted_data)
+    const alert_res = await deleteFromAlerts(db_limit.deleted_data)
```

Comment:
I think there will be problem with this approach of deleting 1 row from probeRequest, notification, and alert in each cycle when user choose to keep logs for all request (`monika --keep-verbose-logs`):

**The amount of data for probeRequest will far exceed alert and notification**

| time | probeRequest | alert | notification |
| ---: | :----------: | :---: | :----------: |
|    1 |      v       |       |              |
|    2 |      v       |       |              |
|    3 |      v       |   v   |              |
|    4 |      v       |   v   |              |
|    5 |      v       |       |              |
|    6 |      v       |       |              |
|    7 |      v       |   v   |              |
|    8 |      v       |       |              |
|    9 |      v       |   v   |              |
|   10 |      v       |   v   |              |
|   11 |      v       |       |              |
|   12 |      v       |   v   |              |
|   13 |      v       |   v   |              |
|   14 |      v       |   v   |      v       |
|   15 |      v       |       |              |
|   16 |      v       |       |              |
|   17 |      v       |       |      v       |
|   18 |      v       |   v   |              |
|   19 |      v       |       |              |
|   20 |      v       |       |              |

For example, the table above illustrate the record of data and the time when it happens. When we remove 3 record for each table then 
- the probeRequest table is deleted until time 18
- the alert table is deleted until time 13
- the notification table is now empty

I think the better approach is to delete records of all table up to certain time, in the case above maybe up to time 17

Reply:
done, delete alert and notifications based on probe_requests.